src/exec.cpp

```diff
@@ -979,7 +983,18 @@ void Executor::raise(const std::string &exception, const std::string &info)
         ip = callstack.top();
         callstack.pop();
     }
-    fprintf(stderr, "unhandled exception %s (%s)\n", exception.c_str(), info.c_str());
+    fprintf(stderr, "unhandled exception %s (%s) at address %lu\n", exception.c_str(), info.c_str(), eip);
+    
+    if (debug != nullptr) {
+        auto line = debug->line_numbers.find(eip);
+        auto p = eip;
+        while (line == debug->line_numbers.end()) {
+            line = debug->line_numbers.find(--p);
```

Comment:
I'm pretty sure you can further improve this loop. There are two calls to `.find()`, but there only needs to be one.